### PR TITLE
[1.18] base-images: Use debian-base:v2.1.0 and debian-iptables:v12.1.0 (includes CVE fixes)

### PR DESCRIPTION
**Pull request by justaugustus**
**What type of PR is this?**

/kind cleanup
/sig release
/area release-eng dependency security

**What this PR does / why we need it**:

- Update dependents to use `debian-base:v2.1.0`
- Update dependents to use `debian-iptables:v12.1.0`

(Selective cherry pick of https://github.com/kubernetes/kubernetes/pull/90665, https://github.com/kubernetes/kubernetes/pull/90697, and https://github.com/kubernetes/kubernetes/pull/90782.)

/assign @dims @BenTheElder
cc: @kubernetes/release-engineering
/priority important-soon

**Which issue(s) this PR fixes**:

Tracking issue: https://github.com/kubernetes/kubernetes/issues/58012

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note
- base-images: Use debian-base:v2.1.0 (includes CVE fixes)
- base-images: Use debian-iptables:v12.1.0 (includes CVE fixes)
```

### CODE REVIEW

**Comment by justaugustus** [comment]
Unsurprisingly, bazel is bazel-ing. Keeping track of the commits I have locally:

```
d2a67a28bc93a9b6e2adf9ed134eb2e630a971f9 (HEAD -> 118-base-images, origin/118-base-images) bazel: Update to repo-infra v0.0.4
2b9ed9278fbca88497cf03b34ff398adef900d67 bazel: Update to 2.2.0
c6463e286649b60cc9577429753ec2bed1859dbc deps: Use debian-base:v2.1.0 and debian-iptables:v12.1.0
3ab7b46a30f89c7dd87cdcd7d496c0605b08eb0c build: Alpha-sort dependencies.yaml
e48260eb08c1a0048df137dfdc52710d2b9c99ab build: Add build-image OWNERS to debian-{base,iptables} and pause dirs
```

**Comment by justaugustus** [comment]
/kind cleanup
/sig release
/area release-eng dependency security
/assign @dims @BenTheElder
cc: @kubernetes/release-engineering
/priority important-soon

**Comment by justaugustus** [comment]
@tpepper @dims @kubernetes/release-engineering -- There's a good chance some of these tests are going to fail due to the various hyperkube references on this branch.

We could:
1. Remove references to hyperkube so the images stop getting built
2. Refactor some of the bash to allow a `KUBE_HYPERKUBE_REGISTRY` (`k8s.gcr.io`) variable, which would allow us to differentiate from `KUBE_BASE_IMAGE_REGISTRY` (which is currently set to `us.gcr.io/k8s-artifacts-prod/build-image`)

I'm leaning towards #1, pending the test results.
Let me know what you all think.

**Comment by justaugustus** [comment]
Thank @dims!
For `build/`:
/assign @cblecker @fejta

**Review by hasheddan — APPROVED** [review]
/lgtm

**Review by cpanato — APPROVED** [review]
/lgtm

**Comment by fejta** [comment]
👍 
/uncc

**Comment by tpepper** [comment]
Since this combines cleanup and image version bumping (CVE fixes), I think it would be good to add something like a "(CVE fixes)" parenthetical to the release note.

**Comment by justaugustus** [comment]
> Since this combines cleanup and image version bumping (CVE fixes), I think it would be good to add something like a "(CVE fixes)" parenthetical to the release note.

@tpepper -- Done!

Needs re-lgtm after rebasing in the [kube-cross bump](https://github.com/kubernetes/release/issues/1281).

**Comment by justaugustus** [comment]
Apt package diffs:

```
$ container-diff diff us.gcr.io/k8s-artifacts-prod/build-image/debian-base-amd64:v2.0.0 us.gcr.io/k8s-artifacts-prod/build-image/debian-base-amd64:v2.1.0 --type=apt 

-----Apt-----

Packages found only in us.gcr.io/k8s-artifacts-prod/build-image/debian-base-amd64:v2.0.0: None

Packages found only in us.gcr.io/k8s-artifacts-prod/build-image/debian-base-amd64:v2.1.0: None

Version differences:
PACKAGE             IMAGE1 (us.gcr.io/k8s-artifacts-prod/build-image/debian-base-amd64:v2.0.0)        IMAGE2 (us.gcr.io/k8s-artifacts-prod/build-image/debian-base-amd64:v2.1.0)
-base-files         10.3 deb10u1, 340K                                                                10.3 deb10u3, 340K
-libgnutls30        3.6.7-4, 2.6M                                                                     3.6.7-4 deb10u3, 2.6M
-libidn2-0          2.0.5-1, 279K                                                                     2.0.5-1 deb10u1, 280K
-libsystemd0        241-7~deb10u1, 767K                                                               241-7~deb10u3, 767K
-libudev1           241-7~deb10u1, 258K                                                               241-7~deb10u3, 258K

$ container-diff diff us.gcr.io/k8s-artifacts-prod/build-image/debian-iptables-amd64:v12.0.1 us.gcr.io/k8s-artifacts-prod/build-image/debian-iptables-amd64:v12.1.0 --type=apt 

-----Apt-----

Packages found only in us.gcr.io/k8s-artifacts-prod/build-image/debian-iptables-amd64:v12.0.1: None

Packages found only in us.gcr.io/k8s-artifacts-prod/build-image/debian-iptables-amd64:v12.1.0: None

Version differences:
PACKAGE             IMAGE1 (us.gcr.io/k8s-artifacts-prod/build-image/debian-iptables-amd64:v12.0.1)        IMAGE2 (us.gcr.io/k8s-artifacts-prod/build-image/debian-iptables-amd64:v12.1.0)
-base-files         10.3 deb10u1, 340K                                                                     10.3 deb10u3, 340K
-libgnutls30        3.6.7-4, 2.6M                                                                          3.6.7-4 deb10u3, 2.6M
-libidn2-0          2.0.5-1, 279K                                                                          2.0.5-1 deb10u1, 280K
-libnftnl11         1.1.4-1~bpo10 1, 217K                                                                  1.1.5-1~bpo10 1, 221K
-libssl1.1          1.1.1d-0 deb10u2, 4M                                                                   1.1.1d-0 deb10u3, 4M
-libsystemd0        241-7~deb10u1, 767K                                                                    241-7~deb10u3, 767K
-libudev1           241-7~deb10u1, 258K                                                                    241-7~deb10u3, 258K
```

**Comment by tpepper** [comment]
Ahead of expected OWNERS lgtm/approve's marking as cherry-pick-approved so the set of merge requirements hopefully come together a little faster and we get back to being build-ready.

**Review by hasheddan — APPROVED** [review]
/lgtm

**Review by saschagrunert — APPROVED** [review]
/lgtm

**Pull request merged**
